File: BOJ_Swift/그리디/BOJ_2217_로프.swift
import Foundation

func getMaxLopeProduct() -> Int {
    let n = Int(readLine()!)!
    var lope = [Int]()
    var res = [Int]()
    
    for _ in 0..<n {
        lope.append(Int(readLine()!)!)
    }
    
    lope.sort()
    
    for i in 0..<n {
        res.append(lope[i] * (n - i))
    }
    
    return res.max() ?? 0
}

print(getMaxLopeProduct())

File: BOJ_Python/그리디/BOJ_2217_로프.py
import sys

n = int(sys.stdin.readline())
lope = []
res = []

for i in range(n):
    lope.append(int(sys.stdin.readline()))

lope.sort()
for i in range(n):
    res.append(lope[i] * (n-i))

print(max(res))